tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -29,18 +63,58 @@ func (r *RealDependencies) execute(cmd string, redirectStdout bool, args ...stri
 	fmt.Printf("%s %s\n", cmd, strings.Join(args, " "))
 	externalCommand := exec.Command(cmd, args...)
 	externalCommand.Stdout = os.Stdout
+
+	var stderr bytes.Buffer
 	// TODO Check naming and redirection logic
 	if !redirectStdout {
-		externalCommand.Stderr = os.Stderr
+		externalCommand.Stderr = &stderr
 	}
-	return externalCommand.Run()
+
+	err := externalCommand.Run()
+	// TODO Check naming and redirection logic
+	if err != nil && !redirectStdout {
+		stderrStr := stderr.String()
+
+		if isXTablesCmd(cmd) {
+			// Transform to xtables-specific error messages with more useful and actionable hints.
+			stderrStr = transformToXTablesErrorMessage(stderrStr, err)
+		}
+
+		// Print stderr to os.Stderr by default.
+		fmt.Fprintln(os.Stderr, stderrStr)
+	}
+
+	return err
+}
+
+// isXTablesCmd checks whether the input is a xtables command.
+func isXTablesCmd(cmd string) bool {
+	for _, c := range XTablesCmds {
```

Comment:
Use a set / map lookup instead of iterating every time.

Reply:
Use

> Use a set / map lookup instead of iterating every time.

Updated by using a set, thanks!